Framework/src/CheckRunner.cxx

```diff
@@ -207,9 +212,9 @@ void CheckRunner::init(framework::InitContext& iCtx)
     iCtx.services().get<CallbackService>().set(CallbackService::Id::Reset, [this]() { reset(); });
     iCtx.services().get<CallbackService>().set(CallbackService::Id::Stop, [this]() { stop(); });
 
-    for (auto& check : mChecks) {
-      check.init();
-      updatePolicyManager.addPolicy(check.getName(), check.getUpdatePolicyType(), check.getObjectsNames(), check.getAllObjectsOption(), false);
+    for (auto& tuple : mChecks) {
+      tuple.second.init();
+      updatePolicyManager.addPolicy(tuple.second.getName(), tuple.second.getUpdatePolicyType(), tuple.second.getObjectsNames(), tuple.second.getAllObjectsOption(), false);
```

Comment:
What about the policies existing before the refresh? i don't see them being cleaned up.

Reply:
correct, good catch. fixing